packages/node/src/utils/promise.ts

```diff
@@ -6,3 +6,13 @@ export async function delay(sec: number): Promise<void> {
     setTimeout(resolve, sec * 1000);
   });
 }
+
+export async function timeout(
+  promise: Promise<void>,
+  sec: number,
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => reject(new Error('promise timeout')), sec * 1000);
+    promise.then(resolve).catch(reject);
```

Comment:
This way even a promise is resolved, reject() will still be called.
You should use `Promise.race()` instead.